cmd/goal/application.go

```diff
@@ -928,17 +929,146 @@ func printable(str string) bool {
 	return true
 }
 
+func jsonPrintable(str string) bool {
+	// htmlSafeSet holds the value true if the ASCII character with the given
+	// array position can be safely represented inside a JSON string, embedded
+	// inside of HTML <script> tags, without any additional escaping.
+	//
+	// All values are true except for the ASCII control characters (0-31), the
+	// double quote ("), the backslash character ("\"), HTML opening and closing
+	// tags ("<" and ">"), and the ampersand ("&").
+	var htmlSafeSet = [utf8.RuneSelf]bool{
```

Comment:
would it be better to generate this programatically? Or flip the logic and have a small set of unsafe runes? `unsafe := unsafeSet[c]` would only need the few exception chars set true.

Reply:
I wasn't that creative. I just copied it from the Golang json encoding library. That's how the json library is determining if it's going to encode that as is, or whether it would apply some additional formatting.
unfortunately, the go library keep this array as an internal array, so I had to duplicate it.